### ISSUE TITLE
empty #+roam_tags: are missed; case not retained in non-empty ones

### ISSUE THREAD

**Issue by notuntoward** [org-roam/org-roam#1342]
`M-x org-roam-tag-add` can't find the roam_tags variable if has no tags i.e. it misses "#+roam_tags:".   It does find non-empty roam_tags strings, but it changes the  "#+roam_tags" part from lower to upper case.  

**Empty roam_tags example:**

In this org file:
```
=Deep Learning: A Generic Approach for Extreme Condition Traffic Forecasting=
/Yu, Rose and Li, Yaguang and Shahabi, Cyrus and Demiryurek, Ugur and Liu, Yan (2017)/

A mixture RNN LSTM predicts extreme traffic

* Org-Roam Metadata
#+title:Yu17gnrcTrafFrcstExtrnRNNlstm
#+created: [2017-03-08 Wed]
#+roam_key: cite:Yu17gnrcTrafFrcstExtrnRNNlstm
#+roam_tags:
```

`M-x org-roam-tag-add LSTM` will put `#+ROAM_TAGS: LSTM` at the top of the file.

**Non-empty roam_tags example:  finds it but changes case**

In this org file:

```
=Towards a simple and robust probabilistic solar variability analysis based on transition probability between variability classes of sky state=
/Zhuang, Fuqiang and Salvado, Philippe and Mangione, Jean-Philippe and Pitaval, S{\'e}bastien and Vernay, Christophe and Carriere, Thomas and Gschwind, Beno{\^\i}t and Saint-Drenan, Yves-Marie and Blanc, Philippe (2020)/
 
A one-step-ahead prob forecast based on discrete sky states.  Kind of a discrete copula + HMM.
 
* Org-Roam Metadata
#+title:Zhuang20pvProbSkyState
#+roam_key: cite:Zhuang20pvProbSkyState
#+roam_tags: probabilistic_forecast
```
`M-x org-roam-tag-add solar_radiation`will replace the #+roam_tags string at the bottom of the file with:

```
#+ROAM_TAGS: solar_radiation probabilistic_forecast
```

This is on Windows 10 Emacs 27.1, running org-roam 1.2.3

**Comment by d12frosted** [comment]
Interesting use case. `org-roam-tag-add` (and some other functions) assume that your file specific metadata is is located at the top of file. So if `#+ROAM_TAGS` is not present, it will put it at the beginning of file. We could actually modify it to search whole file for any other metadata, but it was something like an optimisation. Not sure how to proceed here. I didn't find anything in the org specification regarding location of file-wide metadata, so 🤷 To me it sounds weird to have file-level metadata under heading (where you usually put your heading metadata).

Regarding the case change - it's clearly a bug 😸

**Comment by notuntoward** [comment]
Yeah, I couldn't find a specification for file variable location either.  But I did see file variables sprinkled all over the place in some online org-mode files.  And happily, org-roam is also already written to handle that -- except for the case of a `#+roam_tags:` which is empty.

The reason for hiding them under a headline at the end was so that, when I open an org-roam note, the first thing I see is what I wrote about the paper -- no distractions.
